september-util/src/main/java/edu/asu/diging/gilesecosystem/septemberutil/service/impl/SystemMessageHandler.java

```diff
@@ -73,4 +73,30 @@ public void handleError(String msg, Exception exception) {
             logger.error("Could not send request.", e);
         }
     }
+
+    @Override
+    public void handleWarning(String msg, Exception exception) {
+        logger.error("The following exception was thrown: " + msg, exception);
+        ISystemMessageRequest request;
+        try {
+            request = requestFactory.createRequest(UUID.randomUUID().toString(), null);
+        } catch (InstantiationException | IllegalAccessException e) {
+            logger.error("Could not create request.", e);
+            return;
+        }
+        request.setApplicationId(applicationId);
+        request.setTitle(msg);
+        request.setMessage(exception.getMessage());
+        request.setMessageType(ISystemMessageRequest.WARNING);
+        StringWriter sWriter = new StringWriter();
+        exception.printStackTrace(new PrintWriter(sWriter));
+        request.setStackTrace(sWriter.toString());
+        request.setMessageTime(ZonedDateTime.now().toString());
+
+        try {
+            requestProducer.sendRequest(request, propertiesManager.getProperty(Properties.KAFKA_TOPIC_SYSTEM_MESSAGES));
+        } catch (MessageCreationException e) {
+            logger.error("Could not send request.", e);
+        }
+    }
```

Comment:
this is exactly the same code as the other method (except the warning type), isn't it? I'm sure you can think of a better way to do this ;)